components/user_settings/advanced/user_settings_advanced.jsx

```diff
@@ -134,6 +142,40 @@ export default class AdvancedSettingsDisplay extends React.Component {
         );
     }
 
+    handleDeactivateAccountSubmit = () => {
+        const userId = UserStore.getCurrentId();
+
+        this.setState({isSaving: true});
+
+        updateActive(userId, false,
+            null,
+            (err) => {
+                this.setState({serverError: err.message});
+            }
+        );
+
+        revokeAllSessions(userId,
+            () => {
+                clientLogout();
+            },
+            (err) => {
+                this.setState({serverError: err.message});
+            }
+        );
+    }
+
+    handleShowDeactivateAccountModal = () => {
+        this.setState({
+            showDeactivateAccountModal: true,
+        });
+    }
+
+    handleDeactivateAccountCancelModal = () => {
```

Comment:
Should this be called something like `handleHideDeactivateAccountModal` to mirror above?

Reply:
done, thanks!